test/e2e/framework.go

```diff
@@ -1142,6 +1148,41 @@ func (data *TestData) GetEncapMode() (config.TrafficEncapModeType, error) {
 	return config.TrafficEncapModeInvalid, fmt.Errorf("antrea-conf config map is not found")
 }
 
+func (data *TestData) getFeatures(confName string, antreaNamespace string) (featuregate.FeatureGate, error) {
+	featureGate := featuregate.NewFeatureGate()
+	if err := featureGate.Add(features.DefaultAntreaFeatureGates); err != nil {
+		return nil, err
+	}
+	cfgMap, err := data.GetAntreaConfigMap(antreaNamespace)
+	if err != nil {
+		return nil, err
+	}
+	var agentCfg interface{}
+	if err := yaml.Unmarshal([]byte(cfgMap.Data[confName]), &agentCfg); err != nil {
+		return nil, err
+	}
+	rawFeatureGateMap, ok := agentCfg.(map[interface{}]interface{})["featureGates"]
+	if !ok || rawFeatureGateMap == nil {
+		return featureGate, err
+	}
+	featureGateMap := make(map[string]bool)
+	for k, v := range rawFeatureGateMap.(map[interface{}]interface{}) {
+		featureGateMap[k.(string)] = v.(bool)
+	}
```

Comment:
Too bad there isn't a more direct way to do that... at least not without being able to import the `AgentConfig` struct

Reply:
Maybe this can be alleviated once https://github.com/vmware-tanzu/antrea/issues/723 is fixed?

Reply:
One good thing is that I found the variable name is not good in this block. Let me fix it first.

Reply:
> Maybe this can be alleviated once #723 is fixed?

The configmap update will take 1min to take effect on the mounted file, I think this behavior will slow tests down.